pages/nft/iscn/_iscnId.vue

```diff
@@ -241,70 +241,82 @@ export default class NFTTestMintPage extends Vue {
   }
 
   async mounted() {
-    await Promise.all([
-      this.getISCNInfo().catch(err => {
-        // eslint-disable-next-line no-console
-        console.error(err);
-        this.setError('ISCN_NOT_FOUND')
-      }),
-      // eslint-disable-next-line no-console
-      this.getMintInfo().catch(err => console.error(err)),
-    ]);
-    this.getOgImage()
+    try {
+      await Promise.all([
+        this.getISCNInfo(),
+        this.getMintInfo(),
+      ])
+      this.getOgImage()
+    } catch (error) {
+      this.setError(error)
+    }
   }
 
   async doAction() {
-    this.errorType = ''
-    this.balance = await getAccountBalance(this.address) as string
-    if(this.balance === '0'){
-      this.toggleSnackbar(ErrorType.INSUFFICIENT_BALANCE)
-      return
-    }
-    this.isLoading = true
-    /* eslint-disable no-fallthrough */
-    switch (this.state) {
-      case 'create':
-        if (!this.isUserISCNOwner) {
-          this.toggleSnackbar(ErrorType.USER_NOT_ISCN_OWNER)
-          break
-        }
+    try {
+      this.errorType = ''
+      this.balance = await getAccountBalance(this.address) as string
+      if (this.balance === '0'){
+        throw new Error(ErrorType.INSUFFICIENT_BALANCE)
+      }
+      this.isLoading = true
+      /* eslint-disable no-fallthrough */
+      switch (this.state) {
+        case 'create':
+          if (!this.isUserISCNOwner) {
+            throw new Error(ErrorType.USER_NOT_ISCN_OWNER)
+          }
 
-        try {
           if (this.ogImageBlob) {
-            const arweaveFeeInfo = await this.estimateArweaveFee()
-            if (!this.ogImageArweaveId) {
-              if (!this.ogImageArweaveFeeTxHash) { await this.sendArweaveFeeTx(arweaveFeeInfo) }
-              await this.submitToArweave()
+            try {
+              const arweaveFeeInfo = await this.estimateArweaveFee()
+              if (!this.ogImageArweaveId) {
+                if (!this.ogImageArweaveFeeTxHash) { await this.sendArweaveFeeTx(arweaveFeeInfo) }
+                await this.submitToArweave()
+              }
+            } catch (error) {
+              // eslint-disable-next-line no-console
+              console.error(error);
+              // skip uploading image if there is an error
```

Comment:
I think should keep showing the error if `ogImageBlob` exists, unless we have a UI to allow user to skip uploading

Reply:
I think an easier way is to pop up a snackbar to warn user, but a more complete way is to implement a switch fall-through pattern similar to the ISCN registration flow to let users be able to retry?

Reply:
I think popup a snackbar is fine
The issue would be we need to either allow user to retry the call, or explicitly do sth to skip this step, instead of default skipping on error